src/Admin/Users.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Jilt for WooCommerce Promotions
+ *
+ * This source file is subject to the GNU General Public License v3.0
+ * that is bundled with this package in the file license.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://www.gnu.org/licenses/gpl-3.0.html
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@skyverge.com so we can send you a copy immediately.
+ *
+ * @author    SkyVerge
+ * @copyright Copyright (c) 2020, SkyVerge, Inc. (info@skyverge.com)
+ * @license   http://www.gnu.org/licenses/gpl-3.0.html GNU General Public License v3.0
+ */
+
+namespace SkyVerge\WooCommerce\Jilt_Promotions\Admin;
+
+use SkyVerge\WooCommerce\Jilt_Promotions\Handlers\Installation;
+use SkyVerge\WooCommerce\Jilt_Promotions\Handlers\Prompt;
+use SkyVerge\WooCommerce\Jilt_Promotions\Messages;
+use SkyVerge\WooCommerce\Jilt_Promotions\Notices\Notice;
+
+defined( 'ABSPATH' ) or exit;
+
+/**
+ * Handler for the notice shown when the merchant filters users by the Customer role.
+ *
+ * @since 1.1.0-dev.1
+ */
+class Users extends Prompt {
+
+
+	/** @var string the id associated with the message */
+	private $customer_role_message_id = 'users-customers-role';
+
+	/** @var string user page identifier */
+	private $users_screen_id = 'users';
+
+	/** @var string customer role to match with role parameter */
+	private $customer_role = 'customer';
+
+
+	/**
+	 * Renders a Notice object if the users customer role message is enabled.
+	 *
+	 * @since 1.1.0-dev.1
+	 */
+	private function add_admin_notices() {
```

Comment:
```suggestion
	public function add_admin_notices() {
```

callbacks to WordPress hooks cannot be private, only public

when adding a callback that is meant for internal use (most of the times with hooks), you can add an `@internal` tag to phpdoc to signal that

Reply:
> callbacks to WordPress hooks cannot be private, only public
> 
> when adding a callback that is meant for internal use (most of the times with hooks), you can add an `@internal` tag to phpdoc to signal that

Yeah, my bad. Lack of attention. Thanks for pointing that!